lib/internal/fs.js

```diff
@@ -352,9 +352,12 @@ function validateLen(len) {
   if (!isInt32(len)) {
     if (typeof value !== 'number') {
       err = new ERR_INVALID_ARG_TYPE('len', 'number', len);
-    } else {
+    } else if (Number.isInteger(len)) {
```

Comment:
This should actually be `if (!Number.isInteger(len)) {`, otherwise the messages are wrong.